coco_architecture.md

```diff
@@ -0,0 +1,147 @@
+    The trusted component resides in encrypted memory and handles confidential computing, while the untrusted component interfaces with 
+    the operating system and propagates I/O from encrypted memory to the rest of the system. Intel SGX is an example of a process based TEE
+  - **VM based TEEs**  - In this model, memory is encrypted along a traditional VM boundary running on top of a VMM. AMD SEV-SNP, Intel TDX, 
+    IBM Secure Execution and Protected Execution Functionality (PEF) are examples of VM TEEs.
+- **Hypervisor** - this includes hypervisors such as QEMU/KVM, cloud hypervisor, public cloud provider hypervisors etc...  
+
+The **CC services block** contains a number of services (**colored pink**) which are required for creating a holistic CC platform which the customer 
+can then use to build their solution. This includes the following services (we expect to see more services later on):  
+
+- **Attestation service** -  The primary purpose of the attestation service is to validate the evidence provided by the hardware TEE. 
+  This is the verifier as defined in the RATS architecture.
+- **Key Broker Service (KBS)** -  The KBS is the relying party as defined in the RATS architecture.  Following are its primary function:
+  - Receives evidence from the attester (confidential VM or container) via challenge-response protocol
+  - Relay the evidence to the Attestation Service for verification
+  - Apply appraisal policy for the returned Attestation Results to assess the trustworthiness of the attester
+  - Interact with Key Management Service to retrieve the keys and send it to the attester
+- **Key management service** - A service for securely storing, managing and backing up of cryptographic keys used by applications and users.
+- **Image build service** - A service that is used to build confidential container or VM images for end users
+- **Image registry** - A service that is used to store encrypted and/or signed container and VM images required for confidential computing workloads. 
+  Examples: quay.io, docker hub, registries provided by cloud service providers etc.
+
+On the **Confidential VM/Container side** there is this component (**colored green**)
+
+- **Enclave software stack** - The entity responsible for collecting/sending the evidence, getting the encryption keys 
+  and decrypting/running the application. This includes the attestation agent, image management capabilities etc… 
+  
+## Typical Confidential Container Stack using VM based TEEs (TDX/SEV-SNP/SE)
+
+The following diagram shows how a CC solution is used to provide a confidential container to the end user: 
+
+![CC_TEE_container](https://github.com/confidential-containers/documentation/blob/main/images/CC_TEE_container.jpg)
+
+Following is the workflow when deploying a confidential container (Kubernetes POD):
+
+- CC workload preparation
+  - User builds the container image(s)  (podman build -f …)
```

Comment:
Maybe we should use `skopeo` in the example instead of podman.

Reply:
Perhaps we can just saying (using tools such as skopeo of podman) since we aren't intending to give accurate low level commands as part of the architecture overview?